build-tools/packages/build-cli/src/library/commands/base.ts

```diff
@@ -284,3 +289,18 @@ export abstract class BaseCommand<T extends typeof Command>
 		}
 	}
 }
+
+export abstract class BaseCommandWithBuildProject<
+	T extends typeof Command,
+> extends BaseCommand<T> {
+	private _buildProject: IBuildProject | undefined;
+
+	public getBuildProject(repoRoot?: string): IBuildProject {
+		if (this._buildProject === undefined) {
+			const root = repoRoot ?? findGitRootSync();
```

Comment:
I'm a little nervous about calling git here, because it may mean commands don't work outside of Git repos. I suggest using process.cwd() instead - then each command will start searching at the current dir for the closest BuildProject config.